### PR TITLE
Feat/uri rfc2396

### PR DESCRIPTION
**Pull request by Jammyjamjamman**
Implements iss #9 .

### CODE REVIEW

**Comment by Jammyjamjamman** [comment]
2 Questions:

1. Are we happy with the function names? It conforms to `rmw`, but `escape_uri()` seems funny to me because we're handling `uri`'s, not `url`'s. And it doesn't describe what kind of escape we're doing. I'd propose (from most to least favourite) `percent_escape()`, `percent_escape_uri()`, `escape_uri()`. Also do something similar with `uri_unescape()`. Escape+unescape could be replaced with encode+decode e.g. `percent_encode()`.
2. `unescape_uri()` panics if it fails to decode. Perhaps it should return a e.g. `Result<String, std::string::ParseError>` instead?

**Comment by andy5995** [comment]
> 2 Questions:
> 
>1. Are we happy with the function names? It conforms to `rmw`, but `escape_uri()` seems funny to me because we're handling `uri`'s, not `url`'s. And it doesn't describe what kind of escape we're doing. I'd propose (from most to least favourite) `percent_escape()`, `percent_escape_uri()`, `escape_uri()`. Also do something similar with `uri_unescape()`. Escape+unescape could be replaced with encode+decode e.g. `percent_encode()`.

@Jammyjamjamman , That sounds like a fine idea. The names you suggest seem appropriate to me.

Q2 I answered in Discord

File: tests/rmw_utils.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+use rmwrs::utils;
```
Comment (andy5995) [inline tests/rmw_utils.rs]: These 2 tests can go directly into utils.rs, unless you know of any reason not too. I don't have any set guidelines yet about tests, but I think generally if there are only a few tests that it's fine to keep them with the related functions in the related file. When there are a lot of tests, I find it easier to separate them into single tests/file.rs.

**Review by andy5995 — CHANGES_REQUESTED** [review]

**Comment by Jammyjamjamman** [comment]
Ok I think I'll use `percent_encode()` and `percent_decode()` for names.

File: tests/rmw_utils.rs (hunk 1 above)
Comment (Jammyjamjamman) [reply tests/rmw_utils.rs]: Added this change in latest commit.

**Pull request merged**